### PR TITLE
[User Accounts] Separate "My Preferences" into its own module

### LINKED ISSUE #4440: "My Preferences" help content does not reflect its functionality
Since `my_preferences` and `edit_user` are really both the `user_accounts` module, they share help content documentation. The help content describes only the functionality of the edit_user page.

### LINKED ISSUE #6040: A user should always have access to My Preferences
**Describe the bug**
A "user management" permission is needed to edit one's preferences on LORIS since it is the same module as edit_user.

This is silly because it means a user can't update their own password, username, etc. easily.

**To Reproduce**
Steps to reproduce the behavior (attach screenshots if applicable):
1. Create a user with no permissions
2. On the top right, above logout, click My Preferences
3. You will be forbidden

**What did you expect to happen?**
You should be able to access the page whenever.

**Server Environment (if known):**
- LORIS Version: `aces/master`

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Creates a new module "My Preferences" containing code that was previously bundled with User Accounts functionality.

#### Testing instructions (if applicable)

1. On this branch, run the patch and make sure My Preferences still works.

#### Link(s) to related issue(s)

* Resolves #6040 
* Resolves #4440

### CODE REVIEW

File: modules/my_preferences/php/module.class.inc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * {@inheritDoc}
+     *
+     * @param \User $user The user whose access is being checked.
+     *
+     * @return bool whether access is granted
+     */
+    public function hasAccess(\User $user) : bool
+    {
+        return parent::hasAccess($user);
```
Comment (PapillonMcGill) [inline modules/my_preferences/php/module.class.inc]: Shouldn't this always be `true`

Comment (driusan) [reply modules/my_preferences/php/module.class.inc]: No it shouldn't.. anonymous users don't have access to my preferences.

But there's no reason to override the method just to return the parent value either.

File: modules/my_preferences/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# My Preferences
+
+## Purpose
+
+This module allows a user to edit their basic profile information as well as
+customize their notification settings.
+
+## Intended Users
+
+Every user is able to modify their profile information.
```
Comment (PapillonMcGill) [inline modules/my_preferences/README.md]:
```suggestion
Every authenticated user is able to modify their profile information.
```

**Comment by johnsaigle** [comment]
@ridz1208 @driusan This should probably be tagged to go into the release as it fixes Critical Issue #6040

**Comment by driusan** [comment]
I agree, I changed the branch on the GitHub frontend.

File: modules/my_preferences/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+This module allows a user to edit their basic profile information as well as
+customize their notification settings.
+
+## Permissions
+
+A user is always able to access the My Preferences page.
+
+## Configurations
+
+Some settings related to email notifications affect this module.
```
Comment (driusan) [inline modules/my_preferences/README.md]: What settings related to email notification affect this module?

File: modules/my_preferences/README.md (hunk 2 above)
Comment (driusan) [reply modules/my_preferences/README.md]: I agree with @PapillonMcGill

File: modules/my_preferences/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Interactions with LORIS
+
+- Modifications performed on the My Preferences page will be reflected on 
+the user account page when viewing the edited account.
```
Comment (driusan) [inline modules/my_preferences/README.md]: I don't think that's really an interaction worth noting. Any data modified anywhere should be reflected other places that use the data..

**Comment by driusan** [comment]
@kongtiaowang can you review all the test related changes in this PR to make sure they make sense?

File: modules/my_preferences/README.md (hunk 3 above)
Comment (johnsaigle) [reply modules/my_preferences/README.md]: Notification settings for different modules with upload/download. I'll modify this

File: modules/my_preferences/README.md [hunk 5]
```diff
@@ -20,9 +20,5 @@ A user is always able to access the My Preferences page.
 
 ## Configurations
 
-Some settings related to email notifications affect this module.
-
-## Interactions with LORIS
-
-- Modifications performed on the My Preferences page will be reflected on 
-the user account page when viewing the edited account.
+This module allows a user to modify whether or not they are notified by events in other modules,
```
Comment (ridz1208) [inline modules/my_preferences/README.md]: @johnsaigle I thought this section is for configurations that affect this module, not configurations that are affected by it.

not sure though

Comment (driusan) [reply modules/my_preferences/README.md]: Yes.

Comment (johnsaigle) [reply modules/my_preferences/README.md]: Thanks for clarifying. As far as I know there aren't any.

**Pull request merged**